ts-sdk/whirlpool/tests/createPool.test.ts

```diff
@@ -1,5 +1,301 @@
+    const txFee = 15000n; // 3 signing accounts * 5000 lamports
+    const minRentExempt = balanceChange - txFee;
+
+    assertAccountExists(pool);
+    assert.strictEqual(estInitializationCost, minRentExempt);
+    assert.strictEqual(sqrtPrice, pool.data.sqrtPrice);
+    assert.strictEqual(mintA, pool.data.tokenMintA);
+    assert.strictEqual(mintB, pool.data.tokenMintB);
+    assert.strictEqual(SPLASH_POOL_TICK_SPACING, pool.data.tickSpacing);
+  });
+
+  it("Should create splash pool with 1 TE token (without extension)", async () => {
+    const price = 10;
+    const sqrtPrice = priceToSqrtPrice(price, 6, 6);
+
+    const { instructions, poolAddress, estInitializationCost } =
+      await createSplashPoolInstructions(rpc, mintA, mintTEA, price);
+
+    const balanceBefore = await rpc.getBalance(signer.address).send();
+    const maybePool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    assert.strictEqual(maybePool.exists, false);
+
+    await sendTransaction(instructions);
+
+    const pool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    const balanceAfter = await rpc.getBalance(signer.address).send();
+    const balanceChange = balanceBefore.value - balanceAfter.value;
+    const txFee = 15000n; // 3 signing accounts * 5000 lamports
+    const minRentExempt = balanceChange - txFee;
+
+    assertAccountExists(pool);
+    assert.strictEqual(estInitializationCost, minRentExempt);
+    assert.strictEqual(sqrtPrice, pool.data.sqrtPrice);
+    assert.strictEqual(mintA, pool.data.tokenMintA);
+    assert.strictEqual(mintTEA, pool.data.tokenMintB);
+    assert.strictEqual(SPLASH_POOL_TICK_SPACING, pool.data.tickSpacing);
+  });
+
+  it("Should create splash pool with 2 TE tokens (without extensions)", async () => {
+    const price = 10;
+    const sqrtPrice = priceToSqrtPrice(price, 6, 6);
+
+    const { instructions, poolAddress, estInitializationCost } =
+      await createSplashPoolInstructions(rpc, mintTEA, mintTEB, price);
+
+    const balanceBefore = await rpc.getBalance(signer.address).send();
+    const maybePool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    assert.strictEqual(maybePool.exists, false);
+
+    await sendTransaction(instructions);
+
+    const pool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    const balanceAfter = await rpc.getBalance(signer.address).send();
+    const balanceChange = balanceBefore.value - balanceAfter.value;
+    const txFee = 15000n; // 3 signing accounts * 5000 lamports
+    const minRentExempt = balanceChange - txFee;
+
+    assertAccountExists(pool);
+    assert.strictEqual(estInitializationCost, minRentExempt);
+    assert.strictEqual(sqrtPrice, pool.data.sqrtPrice);
+    assert.strictEqual(mintTEA, pool.data.tokenMintA);
+    assert.strictEqual(mintTEB, pool.data.tokenMintB);
+    assert.strictEqual(SPLASH_POOL_TICK_SPACING, pool.data.tickSpacing);
+  });
+
+  it("Should create splash pool with 1 TE tokens (with extension)", async () => {
```

Comment:
nit: maybe instead of `with extension` be explicit about that it is the TE extension